applications/DEM_application/custom_elements/ship_element.cpp

```diff
@@ -45,6 +45,62 @@ namespace Kratos {
         mDragConstantVector[2] = rigid_body_element_sub_model_part[DEM_DRAG_CONSTANT_Z];
     }
     
+    void ShipElement3D::ComputeWaterDragForce() {
+
+        KRATOS_TRY
+
+        const double water_density = 1000;
+        const double drag_coefficient = 0.75; //https://www.engineeringtoolbox.com/drag-coefficient-d_627.html
+        const double water_level = 0.0;
+
+        for (unsigned int i = 0; i != mListOfRigidFaces.size(); ++i) {
+
+            double rigid_face_area = 0.0;
+            Point rigid_face_centroid;
+            array_1d<double, 3> drag_force = ZeroVector(3);
+            array_1d<double, 3> rigid_body_centroid_to_rigid_face_centroid_vector = ZeroVector(3);
+            array_1d<double, 3> drag_moment = ZeroVector(3);
+            size_t RF_size = mListOfRigidFaces[i]->GetGeometry().size();
+            array_1d<double, 3> rigid_face_z_coords_values = ZeroVector(RF_size);
+            unsigned int number_of_RF_nodes_out_of_water = 0;
+
+            for (unsigned int j = 0; j < RF_size; j++) {
+                rigid_face_z_coords_values[j] = mListOfRigidFaces[i]->GetGeometry()[j].Coordinates()[2];
+                if (rigid_face_z_coords_values[j] > water_level) number_of_RF_nodes_out_of_water++;
+            }
+            
+            if (number_of_RF_nodes_out_of_water == RF_size) continue;
+                        
+            array_1d<double, 3> rigid_face_velocity = ZeroVector(3);
+            
+            rigid_face_velocity = mListOfRigidFaces[i]->GetVelocity();
```

Comment:
For optimization purposes, I would recommend:
```
array_1d<double, 3> rigid_face_velocity;
noalias(rigid_face_velocity) = mListOfRigidFaces[i]->GetVelocity();
```
or use `DEM_COPY_SECOND_TO_FIRST ` without noalias.